h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = element.querySelector('.js-search-bar-input');
+    this._dropdown = element.querySelector('.js-search-bar-dropdown');
+    this._dropdownItems = element.querySelectorAll('.js-search-bar-dropdown-item');
+
+    var self = this;
+    var index = -1;
```

Comment:
Is it possible to get rid of this index variable? I think that when the up or down arrow is pressed, the code can find which menu item in the DOM that currently has the "active" CSS class on it, and then the menu item before or after that one, and then call `updateActiveDropdownItem()` passing it the DOM element for the new menu item to be selected. I think the code will be simpler if it doesn't have an `index` int to keep up to date.